File: src/array/lc_41_first_missing_positive/FirstMissingPositive.java
class FirstMissingPositive {
    public int firstMissingPositive(int[] nums) {
        Arrays.sort(nums);

        if (nums[0] > 1) {
            return 1;
        }
        if (nums[nums.length - 1] <= 0) {
            return 1;
        }

        int currmin = 0;
        boolean diff = false;
        int temp = 0;
        for (int i = 0; i < nums.length; i++){
            if (nums[i] > 0) {
                temp = i;
                break;
            }
        }
        if (nums[temp] > 1) {
            return 1;
        }
        for (int i = temp; i < nums.length-1; i++) {
            if (nums[i+1] - nums[i] > 1) {
                currmin = nums[i] + 1;
                diff = true;
                break;
            }
        }

        // If no positive missing integer found, return the next positive integer
        if (!diff) {
            return nums[nums.length - 1] + 1;
        } 
        return currmin;
    }
}

File: src/array/lc_41_first_missing_positive/first_missing_positive.py
class Solution:
    def firstMissingPositive(self, nums: list[int]) -> int:
        nums = [n for n in nums if n > 0]
        if not nums:
            return 1
        nums.sort()
        missing = False
        temp = 1
        for i in range(len(nums) - 1):
            if nums[i+1] - nums[i] > 1:
                temp = nums[i] + 1 #minimum not included
                missing = True
                break
        
        if not missing:
            temp = nums[-1] + 1
        nums = set(nums)
        for n in range(1, temp):
            if n not in nums:
                return n
        return temp